src/fastga_he/gui/layout_generation.py

```diff
@@ -0,0 +1,255 @@
+# This file is part of FAST-OAD_CS23-HE : A framework for rapid Overall Aircraft Design of Hybrid
+# Electric Aircraft.
+# Copyright (C) 2025 ISAE-SUPAERO
+
+import numpy as np
+import scipy.sparse as sci_sparse
+
+
+class HierarchicalLayout:
+    """
+    Implements the hierarchical graph layout algorithm.
+
+    The algorithm has three main phases:
+    1. Layer assignment (places nodes into layers using flow_hierarchy)
+    2. Crossing minimization (reduces edge crossings between layers using Tutte's algorithm)
+    3. Coordinate assignment (positions nodes within layers)
+    """
+
+    def __init__(self, graph, orientation="TB", node_layer_dict=None):
+        """
+        Initialize the hierarchical layout.
+
+        Args:
+            graph: NetworkX DiGraph
+            orientation: 'TB' (top-bottom), 'BT' (bottom-top), 'LR' (left-right), 'RL' (right-left)
+            node_layer_dict: Dictionary mapping nodes to their layer assignment (required)
+        """
+        self.graph = graph
+        self.orientation = orientation
+        self.layers = []
+        self.node_layer = {}
+        self.positions = {}
+        self.node_layer_dict = node_layer_dict
+
+    def _assign_layers(self):
+        """Assign each node to a layer using provided override with flow_hierarchy as fallback."""
+        self.node_layer = {node: int(layer) for node, layer in self.node_layer_dict.items()}
+
+        max_layer = max(self.node_layer.values()) if self.node_layer else 0
```

Comment:
Can something cause node_layer to be None ? If yes, isn't it impactful for the rest of the process ? So shouldn't an error be raised here ?